src/ucs/sys/sys.c

```diff
@@ -1439,3 +1439,21 @@ ucs_status_t ucs_sys_enum_threads(ucs_sys_enum_threads_cb_t cb, void *ctx)
 
     return ucs_sys_readdir(task_dir, &ucs_sys_enum_threads_cb, &param);
 }
+
+ucs_status_t ucs_sys_get_process_starttime(pid_t pid, ucs_time_t *starttime)
+{
+    static const char *filename_fmt = "/proc/%d";
+    char filename[MAXPATHLEN];
+    struct stat stat_buf;
+    int res;
+
+    snprintf(filename, MAXPATHLEN, filename_fmt, (int)pid);
```

Comment:
can we create `filename` during endpoint creation (and cache it in the ep instead of pid)?

Reply:
technically yes, but it is more memory consuming, and there will not be way to get process starttime directly by pid - we have to get it as set of operations...

are we really want it?

Reply:
imo it would be better - we will have less intrusive check. We will have to increase ep size, not the address, should be ok for shm.
@yosefe, wdyt?

Reply:
> imo it would be better - we will have less intrusive check. We will have to increase ep size, not the address, should be ok for shm.
> @yosefe, wdyt?

agree